### PR TITLE
Check invalid node name

### LINKED ISSUE #18448: Having a scene with a node with a " in the name corrupts scene
**Godot version:**

3.0.2

**OS/device including version:**
KDE Neon

**Issue description:**
If you put a " (quote) in the name of a node, and then save the scene, the scene file is corrupted and cannot be read again by Godot.

**Steps to reproduce:**
1. Create new project with root node.
2. Name it "hello" including quotes
3. Save and close the scene
4. Try to load scene again and be greeted with an error: "Error parsing scene file"

### PR DESCRIPTION
**Pull request by volzhs**
Fix #18448

if we don't strip out the invalid letter after renaming node,
we can still save a scene with the invalid letter for a node name.

### CODE REVIEW

**Comment by KidRigger** [comment]
If possible, please also add checking for '@' and ':' (set_name does not allow them.)

**Comment by volzhs** [comment]
working on it.

**Comment by volzhs** [comment]
There are 2 places to check node name validation.
I think it's not good for managing it.
so I changed my PR to check node name in 1 place, and this PR supersedes #18456 
@KidRigger what do you think?

**Review by zombietype — COMMENTED** [review]
Looks good to me. I agree, better solution. Closing my PR.

File: editor/scene_tree_editor.cpp [hunk 1]
```diff
@@ -644,11 +644,12 @@ void SceneTreeEditor::_renamed() {
 	ERR_FAIL_COND(!n);
 
 	String new_name = which->get_text(0);
-	if (new_name.find(".") != -1 || new_name.find("/") != -1) {
+	if (Node::_validate_node_name(new_name) == false) {
```
Comment (akien-mga) [inline editor/scene_tree_editor.cpp]: Just a style nitpick, but we tend to use `!cond` instead of `cond == false`.

Comment (volzhs) [reply editor/scene_tree_editor.cpp]: @akien-mga fixed.

**Comment by hungrymonkey** [comment]
I want to say that the gltf allows nods name with quotes.

should you put errors in the gltf loader or remove all quotes?

**Comment by akien-mga** [comment]
Now that Godot 3.1 has been released, we only plan to cherry-pick critical bug fixes to the 3.0 branch, unless there is very strong support in favor of a specific non-critical issue. Removing cherrypick label for 3.0.

**Pull request merged**